metadata-validator.py

```diff
@@ -0,0 +1,189 @@
+
+SUBMODULES_KEY = "submodules"
+ELEMENTS_KEY = "elements"
+REQUIRES_KEY = "requires"
+CONFLICTS_KEY = "conflicts"
+NAME_KEY = "name"
+MODULE_NAME_KEY = "module_name"
+ASN1FILES_KEY = "asn1Files"
+IMPORTS_KEY = "imports"
+FROM_KEY = "from"
+
+MODULE_NAME_KEY = "module_name"
+METADATA_KEY = "meta_json"
+METADATA_PATH_KEY = "meta_path"
+
+ASN1SCC = "asn1.exe"
+ASN1SCC_ARGS = "-ACN -c"
+ASN1SCC_OUT_PATH = BUILD_PATH
+CC = "gcc"
+CC_ARGS = "-c -pipe -O2 -fPIC -Wall -W"
+CC_OUT_PATH = BUILD_PATH
+CC_SRC_PATH = BUILD_PATH
+AR = "ar"
+AR_ARGS = "cqs"
+AR_OUT_PATH = BUILD_PATH
+
+
+def getRecursiveDependency(element_lib, element):
+    dependencies = []
+    if ASN1FILES_KEY in element:
+        for element_filename in element[ASN1FILES_KEY]:
+            dependencies.append(element[METADATA_PATH_KEY] + element_filename)
+
+            acn_filename = element[METADATA_PATH_KEY] + os.path.splitext(element_filename)[0] + ACN_EXTENSION
+            if os.path.exists(acn_filename):
+                dependencies.append(acn_filename)
+    
+    if IMPORTS_KEY in element:
+        for imported in element[IMPORTS_KEY]:
+            for item in element_lib:
+                if ASN1FILES_KEY in item:
+                    if imported[FROM_KEY] in [os.path.splitext(x)[0] for x in item[ASN1FILES_KEY]]:
+                        dependencies += getRecursiveDependency(element_lib, item) 
+    
+    return dependencies
+
+def logError(path, message, element):
+    path += METADATA_FILENAME
+    filepath = BUILD_PATH + ERRORLOG_FILE
+    print("\033[1;37;49m%s: \033[1;31;49m%s: \033[0;37;49m%s" % (path, message, element))
+    with open(filepath, "a+") as f:
+        f.write("\n" + path + ": " + message + ": " + element)
+
+def errorCheck(error_flag):
+    if error_flag:
+        print("\033[1;31;49mMetadata validation failed")
+        sys.exit(1)
+    else:
+        print("\033[1;32;49mMetadata validation passed\033[0;37;49m")
+
+def main():
+    try:
+        if(os.path.isdir(BUILD_PATH)):
+            shutil.rmtree(BUILD_PATH)
+
+        os.mkdir(BUILD_PATH)
+    except OSError:
+        print("Error when creating directory: %s" (BUILD_PATH))
+        sys.exit(1)
+
+    metadata = []
+    files = glob.glob(os.path.join(os.path.abspath("."), "*/" + METADATA_FILENAME))
+    for filepath in files:
+        with open(filepath) as f:
+            metadata.append({METADATA_KEY: json.load(f), METADATA_PATH_KEY: filepath.replace(METADATA_FILENAME, "")})
+    
+    element_lib = []
+    for meta in metadata:
+        if SUBMODULES_KEY in meta[METADATA_KEY]: 
+            for submodule in meta[METADATA_KEY][SUBMODULES_KEY]:
+                if ELEMENTS_KEY in submodule: 
+                    for element in submodule[ELEMENTS_KEY]:
+                        element[METADATA_PATH_KEY] = meta[METADATA_PATH_KEY]
+                        element[MODULE_NAME_KEY] = meta[METADATA_KEY][NAME_KEY]
+                        element_lib.append(element)
+
+    error_flag = False;
```

Comment:
instead of storing state, try more functional programming, and refactor this function into several smaller functions (always a good thing - function should be no longer than screen) and return conjunction of their result (like `def check(): return checkA() and checkB()`). If you want the "keep going on error" behaviour, still `def check(): isA = checkA() isB = checkB() return isA and isB` might be more readable.